orderwatch: cleanup job crashes with db error

Seeing this failure after running Mesh for 5hrs:

```
{"error":"leveldb: not found","lastUpdatedCutOff":"2019-07-10T18:15:44.597809441Z","level":"panic","msg":"Failed to find orders by LastUpdatedBefore","time":"2019-07-10T18:45:45Z"}
panic: (*logrus.Entry) (0xe8ce40,0xc009d250a0)

goroutine 52 [running]:
github.com/0xProject/0x-mesh/vendor/github.com/sirupsen/logrus.Entry.log(0xc000070180, 0xc0009d3920, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, ...)
	/go/src/github.com/0xProject/0x-mesh/vendor/github.com/sirupsen/logrus/entry.go:239 +0x2ce
github.com/0xProject/0x-mesh/vendor/github.com/sirupsen/logrus.(*Entry).Log(0xc009d25030, 0xc000000000, 0xc0000b7ce0, 0x1, 0x1)
	/go/src/github.com/0xProject/0x-mesh/vendor/github.com/sirupsen/logrus/entry.go:268 +0xed
github.com/0xProject/0x-mesh/vendor/github.com/sirupsen/logrus.(*Entry).Panic(0xc009d25030, 0xc0000b7ce0, 0x1, 0x1)
	/go/src/github.com/0xProject/0x-mesh/vendor/github.com/sirupsen/logrus/entry.go:306 +0x55
github.com/0xProject/0x-mesh/zeroex/orderwatch.(*Watcher).startCleanupWorker.func1(0xc00030dba0)
	/go/src/github.com/0xProject/0x-mesh/zeroex/orderwatch/order_watcher.go:223 +0x4f9
created by github.com/0xProject/0x-mesh/zeroex/orderwatch.(*Watcher).startCleanupWorker
	/go/src/github.com/0xProject/0x-mesh/zeroex/orderwatch/order_watcher.go:200 +0x3f
```

--- a/db/operations.go
+++ b/db/operations.go
@@ -38,7 +38,7 @@ func findWithIterator(info *colInfo, iter iterator.Iterator, models interface{})
 		return err
 	}
 	modelsVal := reflect.ValueOf(models).Elem()
-	for iter.Next() {
+	for iter.Next() && iter.Error() == nil {
 		// We assume that each value in the iterator is the encoded data for some
 		// model.
 		data := iter.Value()
@@ -47,9 +47,6 @@ func findWithIterator(info *colInfo, iter iterator.Iterator, models interface{})
 			return err
 		}
 		modelsVal.Set(reflect.Append(modelsVal, model.Elem()))
-	}
-	if err := iter.Error(); err != nil {
-		return err
 	}
 	return nil
 }

--- a/db/query.go
+++ b/db/query.go
@@ -5,6 +5,8 @@ import (
 	"fmt"
 	"reflect"
 
+	"github.com/syndtr/goleveldb/leveldb"
+
 	"github.com/albrow/stringset"
 	"github.com/syndtr/goleveldb/leveldb/iterator"
 	"github.com/syndtr/goleveldb/leveldb/util"
@@ -194,6 +196,12 @@ func (q *Query) getAndAppendModelIfUnique(index *Index, pkSet stringset.Set, key
 	}
 	pkSet.Add(string(pk))
 	data, err := q.reader.Get(pk, nil)
+	if err == leveldb.ErrNotFound || data == nil {
+		// It is possible that a separate goroutine deleted the model while we were
+		// iterating through the keys in the index. This is not considered an error.
+		// We simply don't include this model in the final results.
+		return nil
+	}
 	if err != nil {
 		return err
 	}

